pybedlite/overlap_detector.py

```diff
@@ -56,6 +81,38 @@
 from pybedlite.bed_source import BedSource
 
 
+class GenomicSpan(Hashable, Protocol):
```

Comment:
Last minute question: should we get rid of `Genomic` which was introduced in this PR, since intervals can be relative to a transcriptome, amplicon, construct, plasmid, extracted sub-region, etc.  This would also shorten the names:
- `GenomicSpan` -> `Span`
- `StrandedGenomicSpan` -> `StrandedSpan`
- `GenericGenomicSpan` -> SpanType`